lib/util/PDF/PDF_service.dart

```diff
@@ -0,0 +1,385 @@
+      List<dynamic> subTitles, Map<String, String> texts);
+  Future<String?> download(
+      List<dynamic> titles, List<dynamic> subTitles, Map<String, String> texts);
+}
+
+class PDFServiceImpl implements PDFService {
+  static Future<Map<String, dynamic>> getPrefsData() async {
+    final SharedPreferences prefs = await SharedPreferences.getInstance();
+    List<String> userSelectionDifficultEvents =
+        prefs.getStringList('userSelectionPersonalPlan-DifficultEvents') ?? [];
+    List<String> userSelectionMakeSafer =
+        prefs.getStringList('userSelectionPersonalPlan-MakeSafer') ?? [];
+    List<String> userSelectionFeelBetter =
+        prefs.getStringList('userSelectionPersonalPlan-FeelBetter') ?? [];
+    List<String> userSelectionDistractions =
+        prefs.getStringList('userSelectionPersonalPlan-Distractions') ?? [];
+    List<String> phoneNames =
+        prefs.getStringList('PhonePageSavedPhoneNames') ?? [];
+    List<String> phoneNumbers =
+        prefs.getStringList('PhonePageSavedPhoneNumbers') ?? [];
+    String username = prefs.getString('name') ?? '';
+    return {
+      'DifficultEvents': userSelectionDifficultEvents,
+      'MakeSafer': userSelectionMakeSafer,
+      'FeelBetter': userSelectionFeelBetter,
+      'Distractions': userSelectionDistractions,
+      'phoneNames': phoneNames,
+      'phoneNumbers': phoneNumbers,
+      'username': username
+    };
+  }
+
+  static List<List<String>> filterEmptyData(List<List<String>> data) {
+    List<List<String>> filtered = [];
+    for (var i = 0; i < data.length; i++) {
+      if (data[i].isEmpty) {
+        continue;
+      }
+      filtered.add(data[i]);
+    }
+    return filtered;
+  }
+
+  static List<String> setPhones(names, numbers) {
+    List<String> temp = [];
+    for (var i = 0; i < names.length; i++) {
+      temp.add(names[i] + ':' + numbers[i]);
+    }
+    return temp;
+  }
+
+  static Future<pw.Image> loadAssetImage(String path) async {
+    // Load an image from assets and convert it to a format usable by the PDF generator
+    final data = await rootBundle.load(path);
+    final bytes = data.buffer.asUint8List();
+    final image = pw.MemoryImage(bytes);
+    return pw.Image(image);
+  }
+
+  static Future<String?> savePdfAndroid(Uint8List pdfData) async {
+    print(pdfData); // Print the PDF data for debugging purposes
+    try {
+      // Open a save file dialog to allow the user to select a location to save the PDF
+      String? outputFile = await FilePicker.platform.saveFile(
+        dialogTitle: 'Please select an output file:', // Dialog title
+        fileName: 'התוכנית שלי.pdf', // Default file name
+        bytes: pdfData, // PDF data to be saved
+      );
+      //If the user cancels the download
+      return outputFile;
+    } catch (e) {
+      print('Error: $e');
+      return null;
+    }
+  }
+
+  static Future<String?> savePdfWeb(List<int> pdfData) async {
+    // Create a Blob object from the PDF data
```

Comment:
why is this not implemented? I see that pdf lib support web

Reply:
This was a very early implementation. I was not in charge of this, and have no idea what it does. I believe that for now, until it is needed and can me implemented AND tested, there is no need have an implementation.